### PR TITLE
add POST functionality for comments, with test suite

### PR DESCRIPTION
**Pull request by LukeHarrisonDev**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -193,4 +193,100 @@ describe("/api/articles/:article_id/comments", () => {
...
+            .post("/api/articles/1/comments")
+            .send(newComment)
+            .expect(201)
+            .then(({body}) => {
+                expect(body.comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
```
Comment (Harrison365) [inline __tests__/server.test.js:212]: You can be more specific about the user and body here :)

File: __tests__/server.test.js [hunk 2]
```diff
@@ -193,4 +193,100 @@ describe("/api/articles/:article_id/comments", () => {
...
+                username: "lurker",
+                body: "Hello world"
+            }
+            return request(app)
+            .post("/api/articles/9999999999999/comments")
+            .send(newComment)
+            .expect(404)
+            .then(({body}) => {
+                expect(body).toEqual({message: "Not found"})
+            })
```
Comment (Harrison365) [inline __tests__/server.test.js:254]: Don't need to repeat this test :)

File: __tests__/server.test.js [hunk 3]
```diff
@@ -193,4 +193,100 @@ describe("/api/articles/:article_id/comments", () => {
...
+                body: ""
+            }
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send(newComment)
+            .expect(400)
+            .then(({body}) => {
+                expect(body).toEqual({message: "Bad request"})
+            })
+        })
```
Comment (Harrison365) [inline __tests__/server.test.js:290]: Excellent tests ⭐

File: controllers/comments.controllers.js [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+    const newComment = request.body
+    const articleId = request.params
+    insertComment(newComment, articleId)
+    .then((comment) => {
+        response.status(201).send({comment})
+    })
+    .catch((error) => {
+        next(error)
+    })
+}
```
Comment (Harrison365) [inline controllers/comments.controllers.js:24]: Nice 👍

File: models/comments.models.js [hunk 5]
```diff
@@ -14,4 +15,21 @@ function fetchComments(id) {
     })
 }
 
-module.exports = {fetchComments}
\ No newline at end of file
+function insertComment(newComment, article_id) {
+    if (Object.keys(newComment).length === 0 || newComment.body.length === 0) {
+        return Promise.reject({status:400, message: "Bad request"})
```
Comment (Harrison365) [inline models/comments.models.js:20]: Should be rejecting length ===1 too???

File: models/comments.models.js [hunk 6]
```diff
@@ -14,4 +15,21 @@ function fetchComments(id) {
...
+    }
+    let sqlString =
+    `INSERT INTO comments (author, body, article_id)
+    VALUES %L
+    RETURNING *`
+
+    const formattedValues = [[newComment.username, newComment.body, article_id.article_id]]
+
+    const queryString = format(sqlString, formattedValues)
```
Comment (Harrison365) [inline models/comments.models.js:29]: Shouldn't need pg-format here as its only the one insertion. $1, $2, $3 should suffice 🙂

File: models/comments.models.js [hunk 7]
```diff
@@ -14,4 +15,21 @@ function fetchComments(id) {
...
+    return db.query(queryString)
+    .then (({rows}) => {
+        return rows[0]
+    })
+}
```
Comment (Harrison365) [inline models/comments.models.js:34]: Nice work overall. Well done ✅

File: __tests__/server.test.js (hunk 1 above)
Comment (LukeHarrisonDev) [reply __tests__/server.test.js:212]: @Harrison365 
When you say user, do you mean the author when it comes back?

If so, would it just check those two, or would you check the whole object?

File: __tests__/server.test.js (hunk 2 above)
Comment (LukeHarrisonDev) [reply __tests__/server.test.js:254]: @Harrison365 
They're different tests, one for SQL "out of range" and one for a number that doesn't exist yet... I've handled them in different ways. Do you reckon I should still get rid?

File: models/comments.models.js (hunk 5 above)
Comment (LukeHarrisonDev) [reply models/comments.models.js:20]: Good spot, cheers.

File: models/comments.models.js (hunk 6 above)
Comment (LukeHarrisonDev) [reply models/comments.models.js:29]: Awesome, refactored in a 'main' commit. Thanks

File: models/comments.models.js (hunk 7 above)
Comment (LukeHarrisonDev) [reply models/comments.models.js:34]: Thank you. :)

**Pull request merged**